src/services/session.service.ts

```diff
@@ -0,0 +1,38 @@
+import Session from "../session/session";
+import { IPromiseToFileSession } from "session/types";
+import { saveSession } from "./redis.service";
+import * as keys from "../session/keys";
+
+/**
+ * Creates a new ptf session and adds a company to the context.
+ * This will override any ptf session that is already in redis.
+ * @param chSession
+ * @param companyNumber
+ */
+const createPromiseToFileSession = async (
+  chSession: Session): Promise<IPromiseToFileSession> => {
+  const ptfSession: IPromiseToFileSession = {
+    company_number: "",
+  };
+  chSession.appendData(keys.PTF_SESSION, ptfSession);
+  await saveSession(chSession);
+  return ptfSession;
+};
+
+const updatePromiseToFileSessionValue = async (chSession: Session, key: string, value: any): Promise<void> => {
```

Comment:
Rather than have the 'generic' update method that takes a key, why not have specific update methods for company number, company details etc. This would then match the specific 'get' method immediately below. Alternatively, could make the get method below more generic by passing in a key. This interface currently uses a mixture of the two approaches, which is not ideal.

Reply:
Using more generic get method, now used in test.